update-format-crau/src/delta_update.rs

```diff
@@ -85,10 +94,71 @@ pub fn get_signatures_bytes<'a>(mut f: &'a File, header: &'a DeltaUpdateFileHead
     Ok(signatures_bytes.unwrap())
 }
 
+// Take a buffer reader, delta file header, manifest as input.
+// Return path to the whole data blobs, including header, manifest.
+pub fn get_header_data<'a>(f: &'a mut BufReader<File>, header: &'a DeltaUpdateFileHeader, manifest: &proto::DeltaArchiveManifest, tmppath: &Path) -> Result<File, Box<dyn Error>> {
+    fs::create_dir_all(tmppath.parent().unwrap())?;
+    let mut outfile = File::create(tmppath)?;
+
+    // Reset to the beginning of the stream, which means the whole buffer including
+    // delta update header as well as manifest. That is because data that must be verified
+    // with signatures start from the beginning. No header.translate_offset(0).
+    //
+    // Payload data structure:
+    //  | header | manifest | data blobs | signatures |
+
+    f.seek(SeekFrom::Start(0))?;
+
+    let data_length = header.translate_offset(manifest.signatures_offset.unwrap());
+
+    let mut header_data = vec![0u8; data_length as usize];
+    f.read_exact(&mut header_data)?;
+
+    outfile.write_all(&header_data)?;
```

Comment:
This still reads the whole update payload to memory. Do we really need this second file copy? We could change `hash_on_disk` to only read until `data_length`.

Reply:
Done with removing get_header_data, and get its length instead to pass to hash_on_disk.